Source/objects.cpp

```diff
@@ -1164,140 +1163,140 @@ void AddShrine(int i)
 		val = GenerateRnd(shrines);
 	} while (!slist[val]);
 
-	Objects[i]._oVar1 = val;
+	shrine._oVar1 = val;
 	if (!FlipCoin()) {
-		Objects[i]._oAnimFrame = 12;
-		Objects[i]._oAnimLen = 22;
+		shrine._oAnimFrame = 12;
+		shrine._oAnimLen = 22;
 	}
 }
 
-void AddBookcase(int i)
+void AddBookcase(Object &bookcase)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
-	Objects[i]._oPreFlag = true;
+	bookcase._oRndSeed = AdvanceRndSeed();
+	bookcase._oPreFlag = true;
 }
 
-void AddBookstand(int i)
+void AddBookstand(Object &bookstand)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	bookstand._oRndSeed = AdvanceRndSeed();
 }
 
-void AddBloodFtn(int i)
+void AddBloodFountain(Object &bloodFountain)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	bloodFountain._oRndSeed = AdvanceRndSeed();
 }
 
-void AddPurifyingFountain(int i)
+void AddPurifyingFountain(Object &purifyingFountain)
 {
-	int ox = Objects[i].position.x;
-	int oy = Objects[i].position.y;
-	dObject[ox][oy - 1] = -(i + 1);
-	dObject[ox - 1][oy] = -(i + 1);
-	dObject[ox - 1][oy - 1] = -(i + 1);
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	int ox = purifyingFountain.position.x;
+	int oy = purifyingFountain.position.y;
+	dObject[ox][oy - 1] = -(purifyingFountain.GetId() + 1);
+	dObject[ox - 1][oy] = -(purifyingFountain.GetId() + 1);
+	dObject[ox - 1][oy - 1] = -(purifyingFountain.GetId() + 1);
+	purifyingFountain._oRndSeed = AdvanceRndSeed();
 }
 
-void AddArmorStand(int i)
+void AddArmorStand(Object &armorStand)
 {
 	if (!armorFlag) {
-		Objects[i]._oAnimFlag = true;
-		Objects[i]._oSelFlag = 0;
+		armorStand._oAnimFlag = true;
+		armorStand._oSelFlag = 0;
 	}
 
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	armorStand._oRndSeed = AdvanceRndSeed();
 }
 
-void AddGoatShrine(int i)
+void AddGoatShrine(Object &goatShrine)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	goatShrine._oRndSeed = AdvanceRndSeed();
 }
 
-void AddCauldron(int i)
+void AddCauldron(Object &cauldron)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	cauldron._oRndSeed = AdvanceRndSeed();
 }
 
-void AddMurkyFountain(int i)
+void AddMurkyFountain(Object &murkyFountain)

```

Comment:
Have a look at #4571, a lot of these functions can be combined.

Reply:
Done